### DISCUSSION TITLE
Calibration using limited views and rerunning SfM

### DISCUSSION THREAD

**Discussion by ghost** [openMVG/openMVG#2234] [Show and tell]
![Train000](https://github.com/openMVG/openMVG/assets/108303308/5678b57a-bdfd-4a87-b6ff-c4217d50ff92)

A while ago I saw a pull request or issue about using limited views to calibrate then rerunning structure from motion and wanted to write down some notes from my tests.

* The ADJUST_FOCAL_LENGTH|ADJUST_DISTORTION intrinsic refinement had the lowest RMSE.
* Out of 10, 20, 30 views, 20 views had the lowest RMSE.
* The INCREMENTAL engine had the lowest RMSE.
* The DIRECT_LINEAR_TRANSFORM resection method had the lowest RMSE.

All views:
```
NORMAL 0.8
0.883956 418/421 initial
0.537356 396/421 rerun 1
0.611932 414/421 rerun 2
0.909511 421/421 rerun 3

NORMAL 0.6
0.891793 421/421 initial
empty      0/421 rerun 1

HIGH 0.8
0.999362 421/421 initial
0.851488  39/421 rerun 1
failed     0/421 rerun 2

HIGH 0.6
0.988301 421/421 initial
0.988292 421/421 rerun 1
0.988097 421/421 rerun 2
0.988278 421/421 rerun 3
```

20 views:
```
NORMAL 0.8
0.632922 412/421 initial
0.981937 421/421 rerun 3

NORMAL 0.8 DIRECT_LINEAR_TRANSFORM
empty      0/421 initial

NORMAL 0.6 DIRECT_LINEAR_TRANSFORM
0.951345 420/421 initial

HIGH 0.6
0.988924 421/421 initial
0.987903 421/421 rerun 3

HIGH 0.6 DIRECT_LINEAR_TRANSFORM
0.987672 421/421 initial
```

**Comment by pmoulon** [comment]
Thank you for sharing your experiment and feedback to the community!

**Comment by ghost** [comment]
Adding on, three reruns of P3P_NORDBERG_ECCV18 sometimes failed.

All views for an underwater shipwreck dataset:

    P3P_NORDBERG_ECCV18 2 hours - dislocated and curled reconstruction
    DIRECT_LINEAR_TRANSFORM 6 hours - good reconstruction
20 views using INCREMENTALV2 engine:

     DIRECT_LINEAR_TRANSFORM - good partial reconstruction.
Looking forward to P2Pt_FABBRI_ECCV12.
